### ISSUE TITLE
Stops provisioning slaves falsely reporting to jenkins that instance limit reached.

### ISSUE THREAD

**Issue by Arunsrinivask** [KostyaSha/yet-another-docker-plugin#145]
Hello, 

Thank you very much for maintaining this plugin actively, Please help me with the issue below. 

Docker version : 1.11.2
Jenkins version : 2.51
Swarm version : 1.2.5
Yet another docker plugin version: 0.1.0-rc33

Plugin stops provisioning the slaves saying that instance limit has reached even though the actual number of containers in Swarm is far less than the instance limit of label and total limit configured in Jenkins. 
**INFO: Not Provisioning 'test'. Instance limit of '100' reached on server 'swarm'**

Also the number which is reported when it is provisioning slaves is always higher than the actual number of containers of that label running in swarm. 
**INFO: Provisioning 'test' number '36' on 'swarm'; Total containers: '67'**  here both numbers 36 and 67 are higher than they should be.

**Comment by KostyaSha** [comment]
Did you remove manually some containers? I guess i have old logic where local counters are not syncing remote container numbers. Maybe some CachedTtl class requires...

**Comment by Arunsrinivask** [comment]
Yes we are deleting some containers manually, We have use case where users are creating Jenkins jobs which create and delete containers on same hosts used for swarm. Also I have a cron job which deletes all the stopped containers on all swarm hosts, This job is setup as I see that some containers go into exited state before connecting to swarm and slave which got created on Jenkins will be offline and gets destroyed in a minute but not the container in swarm. I am not sure what is causing this, For now I have cronjob to cleanup the containers left out by this issue. I can provide you more information on this after digging more. 
When do you think it is possible to make these local counters sync with remote container numbers? or is there any workaround to do this manually from time to time.

**Comment by KostyaSha** [comment]
I think we can firstly try add synchronised method with counting synchronisation that you can call from groovy if it solve issues, then think about periodic recalculation. Calculating remote container number sounds not performance great, though optionally it could be done with time measuring...

**Comment by joshdick** [comment]
I am also running into this problem; YADP is falsely reporting that instance limits have been reached after manually deleting containers, even after restarting Jenkins.

Is there a manual workaround?

**Comment by KostyaSha** [comment]
Container number need to be updated periodically i guess.

**Comment by KostyaSha** [comment]
One of the potential issues is that counter (hashmap) is stored in DockerCloud and when you press save global configuration in jenkins it recreates this object. 
Describable should be ReconfigurableDescribable. The other possible issue is that because of performance counter wasn't refreshed....
Counting design needs review.
